addons/html_editor/static/src/core/clipboard_plugin.js

```diff
@@ -385,53 +401,52 @@ export class ClipboardPlugin extends Plugin {
+        // Force inline nodes at the root of the container into separate `baseContainers`
         // elements. This is a tradeoff to ensure some features that rely on
         // nodes having a parent (e.g. convert to list, title, etc.) can work
         // properly on such nodes without having to actually handle that
         // particular case in all of those functions. In fact, this case cannot
         // happen on a new document created using this editor, but will happen
         // instantly when editing a document that was created from Etherpad.
+        wrapInlinesInBlocks(container, {
+            baseContainerNodeName:
+                closestBaseContainer?.nodeName ||
+                this.dependencies.baseContainer.getDefaultNodeName(),
+        });
         const result = this.document.createDocumentFragment();
-        let p = this.document.createElement("p");
-        for (const child of childNodes(container)) {
-            if (isBlock(child)) {
-                if (p.hasChildNodes()) {
-                    result.appendChild(p);
-                    p = this.document.createElement("p");
-                }
-                result.appendChild(child);
-            } else {
-                p.appendChild(child);
-            }
+        result.replaceChildren(...childNodes(container));
 
-            if (p.hasChildNodes()) {
-                result.appendChild(p);
-            }
-
-            // Split elements containing <br> into seperate elements for each line.
-            const brs = result.querySelectorAll("br");
-            for (const br of brs) {
-                const block = closestBlock(br);
-                if (
-                    ["P", "H1", "H2", "H3", "H4", "H5", "H6"].includes(block.nodeName) &&
-                    !block.closest("li")
-                ) {
-                    // A linebreak at the beginning of a block is an empty line.
-                    const isEmptyLine = block.firstChild.nodeName === "BR";
-                    // Split blocks around it until only the BR remains in the
-                    // block.
-                    const remainingBrContainer = this.dependencies.split.splitAroundUntil(
-                        br,
-                        block
-                    );
-                    // Remove the container unless it represented an empty line.
-                    if (!isEmptyLine) {
-                        remainingBrContainer.remove();
-                    }
+        // Split elements containing <br> into separate elements for each line.
+        const brs = result.querySelectorAll("br");
+        for (const br of brs) {
+            const block = closestBlock(br);
+            if (
+                (isParagraphRelatedElement(block) ||
+                    this.dependencies.baseContainer.isCandidateForBaseContainer(block)) &&
+                // TODO ? specific exception for "PRE" to keep everything inside one PRE.
```

Comment:
Is this TODO still relevent? It has a question mark so it's rather confusing... ^^'

Reply:
Was in the wrong commit, I moved it in the "PRE" commit instead and removed the ?